admin39/util/model_http_pusher.py

```
@@ -0,0 +1,63 @@
+
+class ModelHttpPusher:
+    class FailedHttpResponseError(RuntimeError):
+        pass
+
+    @staticmethod
+    @shared_task
+    def call(model_resolvable, **options):
+        options = options if options is not None else {}
+        model = options.pop('model')
+        record = ModelHttpPusher.get_object_by_id(model, model_resolvable)
+        record_dict = model_to_dict(record)
+        body = {'type': model}
+        body['payload'] = record_dict
+        url = options.get("url")
+        # secret = options.get("secret")
+        # authenticity_token = options.get("authenticity_token")
+        # authorization_header = cls.get_authorization_header(cls.body_for_signature(body), url, secret=secret, authenticity_token=authenticity_token)
+        authorization_header = "some_auth_hedaer"
+
+        return ModelHttpPusher.api_call(url, json.dumps(body, default=datetime_serializer), authorization_header)
+
+    @staticmethod
+    def api_call(url, body, authorization_header):
+        headers = {"Content-Type": "application/json", "Accepts": "application/json", "Authorization": authorization_header}
+        response = request('POST', url, headers=headers, data=body)
+        if not (200 <= response.status_code < 300):
+            raise ModelHttpPusher.FailedHttpResponseError(f"Host {url} returned HTTP code: {response.status_code}")
+
+        try:
+            json_response = response.json()
+        except json.JSONDecodeError:
+            json_response = {}
+
+        return {
+            "status": response.status_code,
+            "body": json_response
+        }
+
+
+    @staticmethod
+    def get_object_by_id(model, model_resolvable):
+        model_class = apps.get_model(app_label='admin39', model_name=model)
+
+        if model_class is not None:
+            try:
+                obj = model_class.objects.get(id=model_resolvable)
+                return obj
+            except model_class.DoesNotExist:
+                return None
```

Need to raise an exception here, we can't return `None` as object